web/client-api/src/main/java/io/deephaven/web/client/ide/IdeConnection.java

```diff
@@ -7,52 +7,55 @@
 import io.deephaven.web.client.api.QueryConnectable;
 import io.deephaven.web.client.api.WorkerConnection;
 import io.deephaven.web.client.api.console.JsVariableChanges;
+import io.deephaven.web.client.fu.JsLog;
 import io.deephaven.web.shared.data.ConnectToken;
 import io.deephaven.web.shared.fu.JsConsumer;
 import io.deephaven.web.shared.fu.JsRunnable;
 import jsinterop.annotations.JsConstructor;
 import jsinterop.annotations.JsIgnore;
-import jsinterop.annotations.JsMethod;
-import jsinterop.annotations.JsOptional;
-import jsinterop.annotations.JsPackage;
 import jsinterop.annotations.JsType;
 import jsinterop.base.JsPropertyMap;
 
-import java.nio.charset.StandardCharsets;
-
 /**
  */
 @JsType(namespace = "dh")
 public class IdeConnection extends QueryConnectable<IdeConnection> {
-    @JsMethod(namespace = JsPackage.GLOBAL)
-    private static native String atob(String encodedData);
-
-    private static AuthTokenPromiseSupplier getAuthTokenPromiseSupplier(IdeConnectionOptions options) {
-        ConnectToken token = null;
-        if (options != null && options.authToken != null) {
-            token = new ConnectToken();
-            token.setBytes(atob(options.authToken).getBytes(StandardCharsets.UTF_8));
-        }
-        return AuthTokenPromiseSupplier.oneShot(token);
-    }
-
+    private final JsRunnable deathListenerCleanup;
     private final String serverUrl;
 
-    private final JsRunnable deathListenerCleanup;
+    private final ConnectToken token = new ConnectToken();
+
+    @JsConstructor
+    public IdeConnection(String serverUrl, Boolean fromJava) {
```

Comment:
I think need to mark as `fromJava` as `@JsOptional` or else GWT will complain before we even get to the warning below.

Reply:
No, there is no runtime typecheck, but I'll add it, its better to have it. I'm using this to cheat my way around js constructor semantics, and will remove this in 2ish releases